src/Domains.jl

```diff
@@ -48,9 +47,19 @@ function coordinates(domain::RootDomain{FT}) where {FT}
     return domain.compartment_heights
 end
 
+"""
+   SurfaceDomain{FT} <: AbstractDomain{FT}
 
-### Another example, but a ClimaCore FD Column space/domain may be general enough
-### that we keep it as a concrete type of AbstractDomain, rather than an AbstractSoilDomain
+A placeholder for a 2d domain that will coincide with the 
+surface of the soil domain.
+"""
+struct SurfaceDomain{FT} <: AbstractDomain{FT}
+    height::Vector{FT}
```

Comment:
Can you explain this a bit and its use cases? I'm a little confused that a 2D surface domain has a `height` field.

Reply:
hm, so surface variables are defined on the 2d surface `(x,y, z = z_sfc(x,y))`, where `height = z_sfc` (I can rename `height` if we keep this).  There will be topography eventually. 

I think there are two questions here.
1. Does the model require this `z` info to compute RHSs? For example,  the soil model needs topography, `z_sfc(x,y)`, and should have a warped surface to the domain, but Im not sure the actual surface models need it because I dont think `z` shows up in the RHS terms for them. *As long as the correct surface variables are linked up to the right column of soil (in terms of horizontal location - I dont know how we will do this in 3D LSMs), we may not need the surface coordinates at all in compute RHS's.*

2. We initialize variables by computing `coordinates(domain)` and (`similar(coords)`). So we need something here...whether it is `(x,y)` or `(x,y, z_sfc(x,y))`. But right now we are only working with columns, so we only have a notion of `z`?

Reply:
for the first question - we can use a Plane Domain, and give it a single element and single nodal point -> single set of (x,y) is returned by coordinates(model)

The second question is still pertinent! It makes sense to return coordinates as (x,y) pair, but then we cant do similar(coordinates), because coordinates acts like a NamedTuple. this also came up with the 3d domain for soil - returning (x,y,z) coordinates, but wanting similar(coords.z) e.g.

Reply:
fixed by doing
zero_state = map(_ -> zero(FT), coords)
and then do similar(zero_state), instead of similar(coords)